src/cla/tcp/net.rs

```diff
@@ -75,26 +94,53 @@ impl TcpClPacket {
                 MessageType::XferSegment => {
                     let flags = XferSegmentFlags::from_bits_truncate(reader.read_u8().await?);
                     let tid: u64 = reader.read_u64().await?;
+                    let mut extensions = Vec::new();
                     if flags.contains(XferSegmentFlags::START) {
-                        let ext_len: u32 = reader.read_u32().await?;
+                        let mut ext_len: u32 = reader.read_u32().await?;
+                        // parse bundle ids that are request
                         if ext_len != 0 {
-                            warn!("transfer extension are unsupported, ignoring them!");
-                        }
-                        for _ in 0..ext_len {
-                            reader.read_u8().await?;
+                            debug!("parsing transfer extensions");
+                            while ext_len >= 3 {
+                                let flag = reader.read_u8().await?;
+                                let item_type = reader.read_u16().await?;
+                                let item_length = reader.read_u16().await?;
+                                ext_len = ext_len - 3 - item_length as u32;
```

Comment:
there should be a constant and a comment for the number 3 which is used everywhere